vertx-pojo-mapper-common/src/main/java/de/braintags/vertx/jomnigate/observer/impl/JsonSerializationObserver.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * #%L
+ * vertx-pojo-mapper-common
+ * %%
+ * Copyright (C) 2017 Braintags GmbH
+ * %%
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * #L%
+ */
+package de.braintags.vertx.jomnigate.observer.impl;
+
+import java.util.UUID;
+
+import de.braintags.vertx.jomnigate.observer.IObserverContext;
+import de.braintags.vertx.jomnigate.observer.IObserverEvent;
+import de.braintags.vertx.util.exception.ParameterRequiredException;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.Json;
+
+/**
+ * This observer serializes records as json file into a defined directory. It does not return a Future, so that
+ * the caller must not wait for the result. Occuring errors are logged. Per record one file is created.
+ * 
+ * @author Michael Remme
+ * 
+ */
+public class JsonSerializationObserver extends AbstractObserver {
+  private static final io.vertx.core.logging.Logger LOGGER = io.vertx.core.logging.LoggerFactory
+      .getLogger(JsonSerializationObserver.class);
+
+  /**
+   * The name of the property, which defines the path of the directory, where the output shall be stored
+   */
+  public static final String DIRECTORY_PROPERTY = "directoryPath";
+
+  private String parentDir;
+
+  /*
+   * (non-Javadoc)
+   * 
+   * @see
+   * de.braintags.vertx.jomnigate.observer.IObserver#handleEvent(de.braintags.vertx.jomnigate.observer.IObserverEvent,
+   * de.braintags.vertx.jomnigate.observer.IObserverContext)
+   */
+  @Override
+  public Future<Void> handleEvent(IObserverEvent event, IObserverContext context) {
+    try {
+      Object source = event.getSource();
+      if (source != null) {
+        Buffer encoded = Buffer.buffer(Json.encode(source));
+        String path = createFileName(event);
+        Vertx vertx = event.getDataStore().getVertx();
+        vertx.fileSystem().writeFile(path, encoded, res -> {
+          if (res.failed()) {
+            checkCreateDirAndRetry(vertx, encoded, path, res.cause());
+          } else {
+            // we did it
+          }
+        });
+      }
+    } catch (Exception e) {
+      LOGGER.error("error on serializing", e);
+    }
+    return null;
```

Comment:
No, cause the interface allows returned null in order to allow fire-and-forget. 

see IObserver.handleEvent:
   * @return the observer can return a valid {@link Future}, where for the surrounding process will wait; or it can
   *         return NULL, which will process the action as fire-and-forget